File: hindex.java
// Time Complexity :O(log n)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no

class Solution {
    public int hIndex(int[] citations) {
        if(citations == null || citations.length == 0) return 0;
        int n = citations.length;
        int low = 0; int high = n-1;
        while(low <= high){
            int m = low + (high - low)/2;
            int d = n- m;
            if(citations[m] == d){
                return d;
            } else if(citations[m] > d){
                high = m -1;
            }else {
                low = m +1;
            }
        }
        return n - low;
    }
}

File: HIndex.cs
using System;
namespace Algorithms
{
    public class HIndex
    {
        public class Solution
        {
            /// Time Complexity : O(n)
            // Space Complexity :O(1)
            // Did this code successfully run on Leetcode :Yes
            // Any problem you faced while coding this :  No
            public int FindHIndex(int[] citations)
            {
                if (citations == null || citations.Length == 0) return 0;
                int idx = citations.Length;
                for (int i = 0; i < citations.Length; i++)
                {
                    int diff = idx - i;
                    if (citations[i] >= diff)
                        return diff;

                }
                return 0;
            }
        }
    }
}
